### PR TITLE
Fixed a number of value formatting errors.  Removed some code that evaluates differently in browser and Rhino.

### PR DESCRIPTION
**Pull request by ssirowy**
This code is a mess, and I still quite understand some things down the chain in what the evaluator does,  but I was able to include a fix that now fixes all of the issues being reported in helltool.

1.  Fixes Euro formatting as defined by this issue.  Charts no longer report invalid format string issues:

screen shot 2018-07-23 at 2 54 51 pm

2.  Maintains that the dollar sign still get puts in right spot for negative numbers 

screen shot 2018-07-23 at 2 56 48 pm

3.  [This is now fixed as part of this PR as well](https://github.com/SheetJS/ssf/issues/19).  There was a test in master that was failing that now passes:

screen shot 2018-07-23 at 2 58 02 pm

screen shot 2018-07-23 at 2 58 42 pm 2

4.  I've also added a half dozen or so tests to cover some of the currency cases a bit better.

### CODE REVIEW

**Comment by ssirowy** [comment]
As part of this, im going to see if I can make this more automatic to pull into helltool, so please hold off.

**Comment by ssirowy** [comment]
If there is anyone else that should look at this, feel free to include them.

File: package.json [hunk 1]
```diff
@@ -16,6 +16,9 @@
 		"blanket": "~1.2.3",
 		"@sheetjs/uglify-js":"~2.7.3"
 	},
+       "publishConfig": {
+                "registry": "https://nexusrepo.looker.com/repository/looker-npm-private/"
+        },
```
Comment (RichardCzechowski) [inline package.json:21]: I've never touched nexus, so I'll defer to @micahcatlin or @n8agrin here.

File: test/currency.json [hunk 2]
```diff
@@ -7,5 +7,12 @@
...
   [1.99, "$0", "$2"],
-  [1000.40, "$0", "$1000"]
+  [1000.40, "$0", "$1000"],
+  [500, "#,##0\"€\";(#,##0\"€\")", "500€"],
+  [350, "$#,###.00;($#,###.00)", "$350.00"],
+  [-350, "$#,###.00;($#,###.00)", "($350.00)"],
+  [1777, "$#,###.00;($#,###.00)", "$1,777.00"],
+  [-1777, "$#,###.00;($#,###.00)", "($1,777.00)"],
+  [17777, "$#,###.00;($#,###.00)", "$17,777.00"],
+  [-17777, "$#,###.00;($#,###.00)", "($17,777.00)"]
```
Comment (RichardCzechowski) [inline test/currency.json]: These are great. Thanks.

**Review by RichardCzechowski — APPROVED** [review]
Going to say that the tests are doing the reviewing here

File: test/currency.json (hunk 2 above)
Comment (n8agrin) [inline test/currency.json]: I would include another example that was a known error (and has more significant digits) like [these](https://github.com/looker/helltool/issues/32316#issuecomment-332295328)

`[-2084048.35, "$#,##0.00;($#,##0.00)", "($2,084,048.35)"]`

File: test/currency.json [hunk 3]
```diff
@@ -7,5 +7,12 @@
   [1.99999, "$0.00", "$2.00"],
   [1.99, "$0.00", "$1.99"],
   [1.99, "$0", "$2"],
-  [1000.40, "$0", "$1000"]
+  [1000.40, "$0", "$1000"],
+  [500, "#,##0\"€\";(#,##0\"€\")", "500€"],
```
Comment (n8agrin) [inline test/currency.json]: I would also include the literal test from https://github.com/SheetJS/ssf/issues/19#issue-109139459

`[3.14, "#.00\"$\"", "3.14$"]`

**Review by n8agrin — COMMENTED** [review]
Hoping you pulled the sword from the stone here! If so you should consider opening a PR on the public SSF project and try to get this merged upstream!

File: test/currency.json (hunk 3 above)
Comment (ssirowy) [reply test/currency.json]: 👍

File: test/currency.json (hunk 2 above)
Comment (ssirowy) [reply test/currency.json]: 👍

**Review by n8agrin — APPROVED** [review]
🎉 🌮

**Pull request merged**